imagery/i.emissivity/testsuite/test_i_emissivity.py

```diff
@@ -0,0 +1,177 @@
+        for raster in cls.temp_rasters + [cls.output_raster]:
+            cls.runModule("g.remove", type="raster", name=raster, flags="f")
+        cls.del_temp_region()
+
+    def test_emissivity_ndvi_range(self):
+        """Test with NDVI values in the valid range using reference statistics."""
+        self.runModule(
+            "r.mapcalc",
+            expression=f"{self.input_raster} = {self.ndvi_cases['valid_range']}",
+            overwrite=True,
+        )
+        self.assertModule(
+            "i.emissivity",
+            input=self.input_raster,
+            output=self.output_raster,
+            overwrite=True,
+        )
+
+        # Reference statistics for the expected output
+        # These values are derived from the emissivity formula:
+        # For col 1 (NDVI = 0.16): emissivity = 0.97
+        # For col 2 (NDVI = 0.74): emissivity = 0.99
+        # For col 3+ (NDVI = 0.45): emissivity = 0.97 + 0.003 * 0.45 = 0.97135
+        reference_stats = {
+            "mean": 0.97712,  # (0.97 + 0.99 + 0.97135) / 3
+            "min": 0.92,
+            "max": 0.99,
+        }
+        self.assertRasterFitsUnivar(
+            raster=self.output_raster, reference=reference_stats, precision=1e-2
+        )
+
+    def test_complex_mask(self):
+        """Test with a complex mask involving irregular shape and regions."""
+        self.runModule(
+            "r.mapcalc",
+            expression="MASK = if((row() + col()) % 3 == 0 || row() < 3, 1, null())",
+            overwrite=True,
+        )
+        self.temp_rasters.append("MASK")
+
+        self.runModule(
+            "r.mapcalc",
+            expression=f"masked_input = if(isnull(MASK), null(), {self.input_raster})",
+            overwrite=True,
+        )
+        self.temp_rasters.append("masked_input")
+
+        self.assertModule(
+            "i.emissivity",
+            input="masked_input",
+            output=self.output_raster,
+            overwrite=True,
+        )
+
+        self.runModule("r.mask", flags="r")
+
+        reference_stats = {"mean": 0.953836, "min": 0.933356, "max": 0.992236}
+
+        self.assertRasterFitsUnivar(
+            raster=self.output_raster, reference=reference_stats, precision=1e-6
+        )
+
+    def test_emissivity_default_calculation(self):
+        """Test default emissivity calculation with critical NDVI values."""
+        # NDVI_soil = 0.15, NDVI_veg = 0.7, e_soil = 0.96, e_veg = 0.985
+        test_values = [
+            (0.10, 0.96),  # Below NDVI_soil - should use e_soil
+            (0.15, 0.96),  # Exactly NDVI_soil
+            (0.70, 0.985),  # Exactly NDVI_veg
+            (0.80, 0.985),  # Above NDVI_veg - should clamp to e_veg
+            (-0.5, 0.96),  # Invalid NDVI (negative) treated as soil
+            (1.50, 0.985),  # Invalid NDVI (>1) treated as vegetation
+        ]
+
+        for ndvi, expected in test_values:
+            with self.subTest(ndvi=ndvi, expected=expected):
```

Comment:
I think the problem in the CI is related to the usage of subtest, we don't use it anywhere else. So try getting rid of it.